shell.nix

```diff
@@ -0,0 +1,58 @@
+# SPDX-FileCopyrightText: 2022 Google LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+{ pkgs ? import nix/nixpkgs.nix {} }:
+pkgs.mkShell {
+  name = "shell";
+  buildInputs =
+    # Ideally, we'd add all (Hackage) dependencies to the dependency list so we
+    # don't have to compile everything ourselves. I haven't figured out how to
+    # do this yet without introducing conflicts with `cabal.project`.
+    #
+    # pkgs.haskellPackages.bittide-extra.env.nativeBuildInputs ++
+    #
+    [
+      pkgs.buildPackages.cabal-install
+      pkgs.buildPackages.dtc
+      pkgs.buildPackages.gcc
+      pkgs.buildPackages.ghc
+      pkgs.buildPackages.pkg-config
+      pkgs.buildPackages.python310Full
+      pkgs.buildPackages.python310Packages.matplotlib
+      pkgs.buildPackages.python310Packages.scipy
+      pkgs.buildPackages.sbt
+      pkgs.buildPackages.scala
+      pkgs.buildPackages.verilator
+
+      # Rust toolchain. Contains:
+      #
+      #  * rust-analyzer
+      #  * rustc
+      #  * rustdoc
+      #  * rustfmt
+      #  * (.. and more)
```

Comment:
This doesn't mention things like clippy or MIRI (we should probably look into MIRI tests on CI too :eyes: ), is this just the default installation config? Should we have an explicit component list here too maybe?

Reply:
Yeah that sounds nice! Are you talking about [this](https://github.com/rust-lang/miri) MIRI?

Reply:
Yep, that MIRI! It can test (a limited but always growing subset of) unsafe Rust code for undefined behaviour or safety violations! With all the firmware code we have that also uses unsafe, it could be nice to have some checks at some point. But that's for another PR :)

Reply:
I didn't actually answer your question :eyes:

> is this just the default installation config

Not 100% sure. Is Clippy installed by default? If it is it might be less confusing to just leave out the comment :-).

> Should we have an explicit component list here too maybe?

I'm not sure how. There is an `extensions` option, which can contain:

```
       rust-src
       rustc-dev
       rustc-docs
       rust-std
       rust-analysis
       rls-preview
       rust-analyzer-preview
       clippy-preview
       rustfmt-preview
       llvm-tools-preview
       miri-preview
       rust-docs-json-preview
```

..which doesn't seem to match `components`.

Reply:
Since it seems to run all CI tasks successfully it seems like this should be good enough for now!

> Is Clippy installed by default?

There's different profiles, I think the `minimal` one does not come with clippy, but I think the default one does, that's why it all works?

I think the comment can stay, but it confused me a bit more than just assuming it's the default installation :sweat_smile: But I'm fine with this as it is!